Mathlib/CategoryTheory/Closed/PowerObjects.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2025 Klaus Gy. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Klaus Gy
+-/
+import Mathlib.CategoryTheory.Monoidal.Cartesian.Basic
+import Mathlib.CategoryTheory.Topos.Classifier
+/-!
+# Elementary Topos (in Elementary Form)
+
+This ongoing work formalizes the elementary definition of a topos and the direct consequences.
+
+## References
+
+* [S. MacLane and I. Moerdijk, *Sheaves in Geometry and Logic*][MM92]
+-/
+
+universe u v
+
+open CategoryTheory Category Functor Limits MonoidalCategory Opposite
+
+variable {ℰ : Type u} [Category.{v} ℰ] [CartesianMonoidalCategory ℰ]
+
+/-- The covariant functor `B ⊗ [] ⟶ C` from `ℰᵒᵖ` to `Type`. -/
+def WhiskeredHom (B C : ℰ) : ℰᵒᵖ ⥤ Type v :=
+  ⟨ ⟨ fun A ↦ B ⊗ unop A ⟶ C, fun f g ↦ (B ◁ unop f) ≫ g ⟩,
+    fun A ↦ by
+      have : unop (𝟙 A) = 𝟙 (unop A) := by rfl
+      ext; simp[this],
+    fun f f' ↦ by
+      have : B ◁ unop (f ≫ f') = B ◁ unop f' ≫ B ◁ unop f := by aesop_cat
+      ext; simp[this] ⟩
+
+/-- `P` is a power object of `B` if it represents the functor `WhiskeredHom B hc.Ω`. -/
+def IsPowerObjectOf (sc : Classifier ℰ (𝟙_ ℰ)) (B P : ℰ) :=
+  (WhiskeredHom B sc.Ω).RepresentableBy P
```

Comment:
This has to be more general. There should be no reference to `Classifier`, and I believe it should apply to any monoidal category.

Reply:
The alternative, as far as I can see, wolud be this:

```lean
variable {ℰ : Type u} [Category.{v} ℰ] [HasPullbacks ℰ] [CartesianMonoidalCategory ℰ]

noncomputable def WhiskeredSubobject (B : ℰ) : ℰᵒᵖ ⥤ Type (max u v) where
  obj A := Subobject (B ⊗ unop A)
  map f := (Subobject.pullback (B ◁ unop f)).obj
  map_id A := by
    ext1 x
    simp [show unop (𝟙 A) = 𝟙 (unop _) from rfl, Subobject.pullback_id]
  map_comp f f' := by
    ext1 x
    simp [show unop (f ≫ f') = unop f' ≫ unop f from rfl, Subobject.pullback_comp]
```

Is this better, should I continue with this?